media_softlet/agnostic/common/codec/hal/codechal_debug.cpp

```diff
@@ -711,14 +711,9 @@ MOS_STATUS CodechalDebugInterface::DumpRgbDataOnYUVSurface(
         m_osInterface->pfnUnlockResource(m_osInterface, &surface->OsResource);
     }
 
-    if (&(tmpRgbSurface->OsResource))
-    {
-        m_osInterface->pfnFreeResource(m_osInterface, &(tmpRgbSurface->OsResource));
-    }
-    if (tmpRgbSurface)
-    {
-        MOS_Delete(tmpRgbSurface);
-    }
+    m_osInterface->pfnFreeResource(m_osInterface, &(tmpRgbSurface->OsResource));
```

Comment:
NULL check is needed for `tmpRgbSurface->OsResource`
Maybe like:
```
 if (!Mos_ResourceIsNull(&(tmpRgbSurface->OsResource)))
```

Reply:
So, OsResource is not a pointer.
It is data inside tmpRgbSurface.

And because tmpRgbSurface itself is not null (as verified earlier in the function), it points to an existing data member.
The address of that is never null.